spark/src/main/kotlin/com/adevinta/spark/components/progressbar/Progressbar.kt

```diff
@@ -0,0 +1,163 @@
+
+@InternalSparkApi
+@Composable
+internal fun SparkProgressbar(
+    @FloatRange(from = 0.0, to = 1.0) progress: Float,
+    isIndeterminate: Boolean,
+    isRounded: Boolean,
+    intent: ProgressbarIntent,
+    modifier: Modifier = Modifier,
+) {
+    val color = intent.colors().color
+    val trackColor = SparkTheme.colors.onBackground.dim4
+    if (isIndeterminate) {
+        MaterialProgressbar(
+            modifier = modifier.sparkUsageOverlay(),
+            color = color,
+            trackColor = trackColor,
+            strokeCap = if (isRounded) StrokeCap.Round else StrokeCap.Butt,
+        )
+    } else {
+        MaterialProgressbar(
+            modifier = modifier.sparkUsageOverlay(),
+            progress = progress,
+            color = color,
+            trackColor = trackColor,
+            strokeCap = if (isRounded) StrokeCap.Round else StrokeCap.Butt,
+        )
+    }
+}
+
+/**
+ * Determinate Progressbar
+ *
+ * Progressbar express a specified progress.
+ *
+ * @param progress the progress of this progress indicator, where 0.0 represents no progress and 1.0
+ * represents full progress. Values outside of this range are coerced into the range.
+ * @param intent The intent color for the Progressbar.
+ * @param modifier Modifier to be applied to the Progressbar
+ * @param isRounded Controls the border shape of the progressbar. When `true`, this progressbar will have rounded border shape, & the default is rounded
+ */
+
+@Composable
+public fun Progressbar(
+    @FloatRange(from = 0.0, to = 1.0) progress: Float,
+    intent: ProgressbarIntent,
+    modifier: Modifier = Modifier,
+    isRounded: Boolean = true,
+) {
+    SparkProgressbar(
+        progress = progress.coerceIn(0.0f, 1.0f),
+        isIndeterminate = false,
+        intent = intent,
+        modifier = modifier,
+        isRounded = isRounded,
+    )
+}
+
+/**
+ * Intermediate Progressbar
+ *
+ * Progressbar express an unspecified wait time or display the duration of a process.
+ *
+ * @param intent The intent color for the Progressbar.
+ * @param modifier Modifier to be applied to the Progressbar
+ * @param isRounded Controls the border shape of the progressbar. When `true`, this progressbar will have rounded border shape, & the default is rounded
+ */
+
+@Composable
+public fun ProgressbarIndeterminate(
+    intent: ProgressbarIntent,
+    modifier: Modifier = Modifier,
+    isRounded: Boolean = false,
+) {
+    SparkProgressbar(
+        progress = 0f,
+        isIndeterminate = true,
+        intent = intent,
+        modifier = modifier,
+        isRounded = isRounded,
+    )
+}
+
+@Preview(
+    group = "Progressbar",
+    name = "Progressbar",
+)
+@Composable
+internal fun PreviewProgressbar(
+    @PreviewParameter(ThemeProvider::class) theme: ThemeVariant,
+) {
+    PreviewTheme(theme) {
+        ProgressbarIntent.entries.forEach { intent ->
+            val progress: Double = Random.nextDouble(0.0, 1.0)
```

Comment:
I would not use random here as it will generate a different screenshot every time and won't pass the ci